src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/FeatureTagHelperTypeResolver.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// Resolves tag helper types from the <see cref="ApplicationPartManager.ApplicationParts"/>
+    /// of the application.
+    /// </summary>
+    public class FeatureTagHelperTypeResolver : ITagHelperTypeResolver
+    {
+        private readonly ApplicationPartManager _manager;
+
+        /// <summary>
+        /// Initializes a new <see cref="FeatureTagHelperTypeResolver"/> instance.
+        /// </summary>
+        /// <param name="manager">The <see cref="ApplicationPartManager"/> of the application.</param>
+        public FeatureTagHelperTypeResolver(ApplicationPartManager manager)
+        {
+            if (manager == null)
+            {
+                throw new ArgumentNullException(nameof(manager));
+            }
+
+            _manager = manager;
+
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<Type> Resolve(string assemblyName, SourceLocation documentLocation, ErrorSink errorSink)
+        {
+            if (assemblyName == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyName));
+            }
+
+            if (errorSink == null)
+            {
+                throw new ArgumentNullException(nameof(errorSink));
+            }
+
+            var parsedName = new AssemblyName(assemblyName);
+            Assembly assembly;
+            try
+            {
+                assembly = Assembly.Load(parsedName);
+            }
+            catch (Exception ex)
+            {
+                errorSink.OnError(
+                    documentLocation,
+                    string.Format(
+                        "Cannot resolve TagHelper containing assembly '{0}'.Error: {1}",
+                        parsedName.Name,
+                        ex.Message),
+                    assemblyName.Length);
+
+                return Type.EmptyTypes;
+            }
+
+            var feature = new TagHelperFeature();
+            _manager.PopulateFeature(feature);
+
+            var results = new List<Type>();
+            for (int i = 0; i < feature.TagHelpers.Count; i++)
+            {
+                var tagHelper = feature.TagHelpers[i];
+                var tagHelperAssembly = tagHelper.Assembly;
+
+                if (tagHelperAssembly.Equals(assembly))
```

Comment:
Will this fall over with strong naming? We typically are looser with our assembly checks. [Example](https://github.com/dotnet/cli/blob/a81f1c08f0e87f8ee39e3d6e78bcc556bce7f38b/src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs#L47-L73)